tests/test_config.py

```
@@ -254,6 +254,11 @@ def test_concrete_http_class() -> None:
     assert config.http_protocol_class is H11Protocol
 
 
+def test_http2_requires_zttp_protocol() -> None:
+    with pytest.raises(ValueError, match="HTTP/2 requires `http='zttp'`."):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)